docs/eventing/channel-based-broker.md

```diff
@@ -0,0 +1,214 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kafka-channel
+  namespace: knative-eventing
+data:
+  channelTemplateSpec.apiVersion: messaging.knative.dev/v1alpha1
+  channelTemplateSpec.kind: KafkaChannel
+  channelTemplateSpec.specJson: '{"numPartitions": 3, "replicationFactor": 1}'
+```
+
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: config-br-defaults
+  namespace: knative-eventing
+data:
+  default-br-config: |
+    clusterDefault:
+      apiVersion: v1
+      kind: ConfigMap
+      name: imc-channel
+      namespace: knative-eventing
+    namespaceDefaults:
+      test-broker-6:
+        apiVersion: v1
+        kind: ConfigMap
+        name: kafka-channel
+        namespace: knative-eventing
+```
+
+
+## Installing Broker by Annotation
+
+The easiest way to get Broker installed, is to annotate your namespace
+(replace `default` with the desired namespace):
+
+```shell
+kubectl label namespace default knative-eventing-injection=enabled
+```
+
+This will automatically create a `Broker` named `default` in the `default`
+namespace. As per above configuration, it would be configured to use Kafka
+channels.
+
+```shell
+kubectl -n default get broker default
+```
+
+_NOTE_ `Broker`s created due to annotation will not be removed if you remove the
+annotation. For example, if you annotate the namespace, which will then create
+the `Broker` as described above. If you now remove the annotation, the `Broker`
+will not be removed, you have to manually delete it.
+
+For example, to delete the injected Broker from the foo namespace:
+
+```shell
+kubectl -n foo delete broker default
+```
+
+## Installing Broker Manually
+
+In order to setup a `Broker` manually, we must first create the required
+`ServiceAccount`s and give them the proper RBAC permissions. This setup is
+required once per namespace. These instructions will use the `default`
+namespace, but you can replace it with any namespace you want to install a
+`Broker` into.
+
+Create the `ServiceAccount` objects.
+
+```shell
+kubectl -n default create serviceaccount eventing-broker-ingress
+kubectl -n default create serviceaccount eventing-broker-filter
+```
+
+Then give them the needed RBAC permissions:
+
+```shell
+kubectl -n default create rolebinding eventing-broker-ingress \
+  --clusterrole=eventing-broker-ingress \
+  --serviceaccount=default:eventing-broker-ingress
+kubectl -n default create rolebinding eventing-broker-filter \
+  --clusterrole=eventing-broker-filter \
+  --serviceaccount=default:eventing-broker-filter
+```
+
+Note that these commands each use three different objects, all named
+`eventing-broker-ingress` or `eventing-broker-filter`. The `ClusterRole` is
+installed with Knative Eventing
+[here](https://github.com/knative/eventing/blob/master/config/200-broker-clusterrole.yaml).
+The `ServiceAccount` was created two commands prior. The `RoleBinding` is
+created with this command.
+
+Create RBAC permissions granting access to shared configmaps for logging,
```

Comment:
These instructions can be replaced with a ConfigMapPropagation (that's what the namespace controller uses now)

Reply:
Is this something that you or @grac3gao could do after we get the initial changes in?